### ISSUE TITLE
Particles attached to actors

### ISSUE THREAD

**Issue by cxong** [cxong/cdogs-sdl#437]
There should be a type of particle that has their position attached/parented to an actor. This would make the muzzle flashes better - they are drawn slightly offset if the actor is moving - and would be useful for things like #28, #188, #436.

**Comment by sauer2** [comment]
@cxong Are objects and pickups actors, too?
Imagine that you could have smoke and sparks coming from a broken device or weird waves coming from an alien device...

**Comment by cxong** [comment]
No, but a generic way to implement this feature would be to add parent-child relationships between all game objects, which would allow attaching things to objects/pickups.
